src/shared/machine-id-setup.c

```diff
@@ -61,7 +61,7 @@ static int generate_machine_id(const char *root, sd_id128_t *ret) {
                 return 0;
         }
 
-        if (isempty(root) && running_in_chroot() <= 0) {
+        if (empty_or_root(root) && running_in_chroot() <= 0) {
```

Comment:
is this really desirable? in many cases we allow `--root=/` as a mechanism for forcing an "offline" mode, while still operating on the root dir. if we do the getenv_for_pid() thing below I'd claim this is very much an "online" operation, and hence `--root=/` should really disable that.